### ISSUE TITLE
Make components page default show all components

### ISSUE THREAD

**Issue by starfishfive** [DependencyTrack/dependency-track#3719]
### Current Behavior

Displays No matching records found

This is a dealbreaker to comply with CIS Control 16.4

### Proposed Behavior

Show a list of all components.

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this enhancement was already requested

**Comment by valentijnscholten** [comment]
I agree that the search on the Components page is very limited and could use some improvements. But what exact use-case can you currently not perform good enough to comply with CIS Control 6.4? Dependency Track does exactly what is stated in the requirements:

> Establish and manage an updated inventory of third-party components used in development, often referred to as a “bill of materials,” as well as components slated for future use. This inventory is to include any risks that each third-party component could pose. Evaluate the list at least monthly to identify any changes or updates to these components, and validate that the component is still supported.

**Comment by starfishfive** [comment]
@valentijnscholten I would like the posibility to go trough the complete list, to insure that each component has:

1. Risk associated with component (Risk Score)
2. Whether component is supported (Version/Latest Version) - Future enhancement add depecrated true/false column?

As stated by the CIS assessment specification of safeguard 16.4

**Comment by rknj** [comment]
We're also interested into having a way to show all the components (across all projects) to manage the outdated libraries at a higher level.
